src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +24,26 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        Set<Character> uniqueCharsSet = new TreeSet<>();
+        StringBuilder result = new StringBuilder();
+        int maxLength = 5;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            int symbol = reader.read();
+            while (symbol != -1) {
+                if (Character.isLetter(symbol)) {
+                    uniqueCharsSet.add(Character.toLowerCase((char) symbol));
+                }
+                symbol = reader.read();
+            }
+        } catch (IOException e) {
+            throw new FileNotFoundException();
+        }
+        for (char c : uniqueCharsSet) {
+            result.append(c);
+        }
```

Comment:
let's check on each iteration if current length equals 5
if yes, break the loop